activelearning/dataset/grid.py

```diff
@@ -97,44 +91,24 @@ def __init__(
         float_precision=64,
     ):
         super().__init__(
-            float_precision=float_precision, batch_size=batch_size, shuffle=shuffle
+            env=env,
+            float_precision=float_precision,
+            batch_size=batch_size,
+            shuffle=shuffle,
         )
 
-        self.normalize_scores = normalize_scores
         self.train_fraction = train_fraction
         self.train_path = train_path
         self.test_path = test_path
-        self.grid_size = grid_size
 
         self.initialise_dataset()
 
-    def statetorch2readable(self, state, alphabet={}):
+    def proxy2state(self, proxy_state):
```

Comment:
Why/where is this method needed? I'm asking because if we can do things without the need for these method, it would be great. One, this isn't needed in the gflownet repo and hence is not implemented. So it would have to be implemented for each new environment here (not great). Second, in some cases, there will not be a one-to-one mapping from proxy to state.

Reply:
we need it because the sampled results are in proxy format and must be mapped back to environment format for storing. this is not needed in gflownets generally because we don't add any new datapoints to the train set

Reply:
Uhm, but if you sample candidates from a GFlowNet, they will/can be in environment format.

Reply:
interesting! That is the better option then. I just have to figure out how to adopt this for the baseline samplers

Reply:
So to clarify: the environment format is the baseline (main) format, from which the others (proxy, policy, readable) can be obtained. So that one is always available. That's why I would suggest that storage is done in environment format.